### PR TITLE
Add hyperapp-axios

### PR DESCRIPTION
**Pull request by jonlov**

### CODE REVIEW

**Comment by zaceno** [comment]
@jonlov Looks nice! But the live example linked from the readme seems broken? At least I couldn't get anything to happen when I clicked "Visit Repo". (Using Safari)

**Comment by jonlov** [comment]
Thanks @zaceno! I really needed something simple when fetching data from servers without rewriting a lot of code.

I checked the bug and fix it, thanks for letting me know. Also I added a new example and changed the repo name.

**Comment by JorgeBucaran** [comment]
Thank you, @jonlov, for the submission. Could you explain what this library does in layman's terms? I read the README, but I am still not confident I could explain what it does to someone elsee.

**Comment by jonlov** [comment]
@JorgeBucaran basically, it allows you to fetch data and makes it available to you on the state. So it handles the state for you when fetching data.

The problem that I found is that every time I wanted to make a request, I needed to write the whole axios.then( .. [send it to the state] .. ).catch(  .. [send it to the state] .. ) again and again and again (which means more kb of the same). Also, if you're making an app with a lot of requests the project starts getting very messy so I came with this solution.

Probably there is a better way to explain it, but that's basically what it does. Should I add the problem and the solution that hyperapp-axios does to the documentation? Have you ever run into the same issue?

Off the topic, thanks for making hyperapp! I'm really enjoying it. I'm trying to build a mid-size app with it, that's why I created this repo. I'm working on a login/signup/forgot password with OAuth with hyperapp/nodejs while keeping it as minimal as possible. I'll put the repository as well online when its done.

**Comment by JorgeBucaran** [comment]
> @jonlov Probably there is a better way to explain it, but that's basically what it does. Should I add the problem and the solution that hyperapp-axios does to the documentation?

Yes, please. This could go either in a new section: background, rationale, why, backdrop, etc. 

Maybe all you need is a README like Hyperapp's [README](https://github.com/hyperapp/hyperapp/blob/master/README.md) with a cleaner structure. I don't mean your README is unclean, but it could use a bit of work if you know what I mean. 🙏

**Comment by JorgeBucaran** [comment]
> @jonlov Off the topic, thanks for making hyperapp! I'm really enjoying it. I'm trying to build a mid-size app with it, that's why I created this repo. I'm working on a login/signup/forgot password with OAuth with hyperapp/nodejs while keeping it as minimal as possible. I'll put the repository as well online when its done.

Awesome! I am looking forward to it.

**Comment by okwolf** [comment]
I think the repo README should have a bit more than code examples.

**Comment by JorgeBucaran** [comment]
@jonlov It doesn't have to be perfect, but if you could address at least half of the points above, we can merge here. You can add tests later as well if you want.

**Comment by jonlov** [comment]
Thanks for the feedback guys! I'll do all of the points you mention @JorgeBucaran and try to explain it a little bit more detailed. I'll let you guys know when it's finished.

Do you find it helpful? Is it something that you guys would use?

**Comment by zaceno** [comment]
@jonlov For me it's a question of axios itself: I'm not familiar enough with it to know when I'd use it over a plain fetch. But your pattern makes sense, I think -- especially if I need to work with multiple end points, I'd want to avoid repeating the code for the calls. So yes, I see how it adds value.

**Comment by jonlov** [comment]
@zaceno yes, I’m

**Comment by jonlov** [comment]
@zaceno yes, I agree with you. There’s some advantages using axios like interceptors, etc but it all depends what you need. I was thinking to do the same library but with fetch instead of axios or do one library where you can use axios or fetch with the same approach because the purpose is to avoid rewriting the code of the calls.

What do you guys think?

My mistake for closing the pull. I’m using the phone and the close and comment button trick me

**Comment by JorgeBucaran** [comment]
Hi @jonlov. Did you have a chance to brush up the README and documentation?

I made a simple template you can use. 
- https://gist.github.com/JorgeBucaran/533f75c2330cf79afc2e622ad7a07e84

**Comment by jonlov** [comment]
Hi @JorgeBucaran. I have been working on the plugin, it's almost done. This is why I haven't updated the documentation. Give me couple more days and I'll make it happen. Thanks for sharing that template! Documentation is no my thing, but that template will be very helpful.

**Comment by jorgebucaran** [comment]
Anything for us @jonlov? :)

**Comment by jonlov** [comment]
I've been super busy. I will have some free time next week. I will get back to this repo. @jorgebucaran

**Comment by jorgebucaran** [comment]
👋 @jonlov No problem! We're all running busy schedules. Please take your time and come back when you have the time. In the meantime, I'm closing this!
